apis/v1alpha1/httproute_types.go

```diff
@@ -535,26 +535,29 @@ type HTTPRouteForwardTo struct {
 	Port PortNumber `json:"port"`
 
 	// Weight specifies the proportion of traffic forwarded to the backend
-	// referenced by the ServiceName or BackendRef field. computed as
+	// referenced by the ServiceName or BackendRef field. This is computed as
 	// weight/(sum of all weights in this ForwardTo list). Weight is not a
-	// percentage and the sum of weights does not need to equal 100. If only one
-	// backend is specified, 100% of the traffic is forwarded to that backend.
-	// If weight is set to 0, no traffic should be forwarded for this entry.
-	// If unspecified, weight defaults to 1.
+	// percentage and the sum of weights does not need to equal 100.
+	//
+	// If only one backend is specified and it has a weight greater than 0, 100%
+	// of the traffic is forwarded to that backend. If weight is set to 0, no
+	// traffic should be forwarded for this entry. If unspecified, weight
+	// defaults to 1.
 	//
 	// Support: Core
 	//
 	// +kubebuilder:default=1
 	// +kubebuilder:validation:Minimum=0
-	// +kubebuilder:validation:Maximum=10000
+	// +kubebuilder:validation:Maximum=1000000000
 	Weight int32 `json:"weight,omitempty"`
 
 	// Filters defined at this-level should be executed if and only if the
 	// request is being forwarded to the backend defined here.
 	//
-	// Conformance: Filtering support, including core filters, is NOT guaranteed
-	// at this-level. Use Filters in HTTPRouteRule for portable filters across
-	// implementations.
+	// Conformance: Support for filters defined at this level is NOT guaranteed.
+	// Labels implying "Core" or "Extended" support for a specific filter type
+	// do NOT apply at this level. For broader support of filters, use the
+	// Filters field defined one level higher as part of HTTPRouteRule.
```

Comment:
Should we say that the conformance level here is extended or custom?

Reply:
Rob -- I think we should say filters need to be documented conformance "in the context" of where it is used. Therefore, there is a separate conformance level for:

pre backend selection vs post backend selection.

We should modify the docs in filter to reflect this distinction.

This is due to the reuse of types. We could make them different types but with the same content, not clear long-term if that is even a bad idea.

Reply:
I think it would add unnecessary overhead to define filter types twice. Maybe we should simply provide two conformance levels for each filter:
```
Support in HTTPRouteRule: Core
Support in HTTPRouteFilter: Extended
```

Reply:
I like that idea, would this be an understandable way to communicate that? 
 
```
Support in HTTPRouteRule: Core
Support in HTTPRouteFilter: Extended
```

Reply:
lgtm

Reply:
👍 Updated spec to include that.